wallet/grpc/core/src/convert.rs

```diff
@@ -1,10 +1,127 @@
+///
+/// # Arguments
+/// * `tx` - Byte slice containing the domain transaction data
+///
+/// # Returns
+/// * `Result<RpcTransaction, Status>` - Deserialized transaction or error status
+fn deserialize_domain_tx(tx: &[u8]) -> Result<RpcTransaction, Status> {
+    let tx = TransactionMessage::decode(tx).map_err(|err| Status::invalid_argument(err.to_string()))?;
+    RpcTransaction::try_from(tx)
+}
+
+/// Extracts and deserializes a partially signed transaction from bytes into an RpcTransaction.
+///
+/// # Arguments
+/// * `tx` - Byte slice containing the partially signed transaction data
+///
+/// # Returns
+/// * `Result<RpcTransaction, Status>` - Deserialized transaction or error status
+fn extract_tx(tx: &[u8], ecdsa: bool) -> Result<RpcTransaction, Status> {
+    let tx = PartiallySignedTransaction::decode(tx).map_err(|err| Status::invalid_argument(err.to_string()))?;
+    let tx_message = extract_tx_deserialized(tx, ecdsa).map_err(|err| Status::invalid_argument(err.to_string()))?;
+    RpcTransaction::try_from(tx_message)
+}
+
+/// Extracts and processes a partially signed transaction into a regular transaction message.
+/// Handles both single-signature and multi-signature inputs, constructing appropriate signature scripts.
+fn extract_tx_deserialized(mut partially_signed_tx: PartiallySignedTransaction, ecdsa: bool) -> Result<TransactionMessage, Error> {
+    if let Some(tx) = partially_signed_tx.tx.as_mut() {
+        for (i, input) in partially_signed_tx.partially_signed_inputs.iter().enumerate() {
+            let is_multi_sig = input.pub_key_signature_pairs.len() > 1;
+            if is_multi_sig {
+                let mut script_builder = &mut ScriptBuilder::new();
+                let mut signature_counter = 0;
+                for pair in input.pub_key_signature_pairs.iter() {
+                    script_builder = script_builder.add_data(pair.signature.as_slice())?;
+                    signature_counter += 1;
+                }
+
+                if signature_counter < input.minimum_signatures {
+                    return Err(Error::MissingSignatures(input.minimum_signatures - signature_counter));
+                }
+
+                let redeem_script = partially_signed_input_multisig_redeem_script(input, ecdsa)?;
+                script_builder = script_builder.add_data(redeem_script.as_slice())?;
+                let sig_script = script_builder.script();
+                tx.inputs[i].signature_script = Vec::from(sig_script);
+            } else {
+                if input.pub_key_signature_pairs.is_empty() {
+                    return Err(Error::MissingSignature);
+                }
+                let mut script_builder = ScriptBuilder::new();
+                let sig_script = script_builder.add_data(input.pub_key_signature_pairs[0].signature.as_slice())?.script();
+                tx.inputs[i].signature_script = Vec::from(sig_script);
+            }
+        }
+        return Ok(tx.clone());
```

Comment:
if u accept owned object, u can remove `as_mut` and clone at the end

Reply:
fixed